crates/next-core/src/middleware.rs

```diff
@@ -139,23 +166,52 @@ impl Issue for MiddlewareMissingExportIssue {
     }
 
     #[turbo_tasks::function]
-    fn title(&self) -> Vc<StyledString> {
-        let file_name = self.file_path.file_name();
-
-        StyledString::Line(vec![
-            StyledString::Text(rcstr!("The ")),
-            StyledString::Code(self.file_type.clone()),
-            StyledString::Text(rcstr!(" file \"")),
-            StyledString::Code(format!("./{}", file_name).into()),
-            StyledString::Text(rcstr!("\" must export a function named ")),
-            StyledString::Code(format!("`{}`", self.function_name).into()),
-            StyledString::Text(rcstr!(" or a default function.")),
-        ])
-        .cell()
+    async fn title(&self) -> Result<Vc<StyledString>> {
+        let relative_path_str = self.get_relative_path();
```

Comment:
Just use `self.file_path.value_to_string().await?` for the message. That's what the other errors do as well.

That `get_relative_path` doesn't work for us, it uses `env::current_dir` which causes nondeterministic cache content.